File: specification/communication/resource-manager/Microsoft.Communication/preview/2021-10-01-preview/examples-java/communicationServices/listKeys.java
import com.azure.core.util.Context;

/** Samples for CommunicationServices ListKeys. */
public final class Main {
    /*
     * x-ms-original-file: specification/communication/resource-manager/Microsoft.Communication/preview/2021-10-01-preview/examples/communicationServices/listKeys.json
     */
    /**
     * Sample code: List keys.
     *
     * @param manager Entry point to CommunicationManager.
     */
    public static void listKeys(com.azure.resourcemanager.communication.CommunicationManager manager) {
        manager
            .communicationServices()
            .listKeysWithResponse("MyResourceGroup", "MyCommunicationResource", Context.NONE);
    }
}

File: specification/communication/resource-manager/Microsoft.Communication/preview/2021-10-01-preview/examples-js/communicationServices/listKeys.js
const { CommunicationServiceManagementClient } = require("@azure/arm-communication");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Get the access keys of the CommunicationService resource.
 *
 * @summary Get the access keys of the CommunicationService resource.
 * x-ms-original-file: specification/communication/resource-manager/Microsoft.Communication/preview/2021-10-01-preview/examples/communicationServices/listKeys.json
 */
async function listKeys() {
  const subscriptionId = "12345";
  const resourceGroupName = "MyResourceGroup";
  const communicationServiceName = "MyCommunicationResource";
  const credential = new DefaultAzureCredential();
  const client = new CommunicationServiceManagementClient(credential, subscriptionId);
  const result = await client.communicationServices.listKeys(
    resourceGroupName,
    communicationServiceName
  );
  console.log(result);
}

listKeys().catch(console.error);
